engine/src/vault/crypto_box.rs

```diff
@@ -149,3 +151,154 @@ pub trait Decrypt<T: TryFrom<Vec<u8>>>: AsRef<[u8]> {
         T::try_from(opened).map_err(|_| DecryptError::Invalid)
     }
 }
+
+//####### NON CONTIGUOUS KEY
+
+/// A key to the crypto box.  [`NCKey`] is stored on the heap which makes it easier to erase. Makes use of the
+/// [`NonContiguousMemory`] type to protect the data.
+#[derive(Serialize, Deserialize)]
+pub struct NCKey<T: BoxProvider> {
+    /// the guarded raw bytes that make up the key
+    pub key: NonContiguousMemory,
+
+    /// phantom data to call to the provider.
+    #[serde(skip_serializing, skip_deserializing)]
+    _box_provider: PhantomData<T>,
+}
+
+impl<T: BoxProvider> NCKey<T> {
+    /// generate a random key using secure random bytes
+    #[allow(dead_code)]
```

Comment:
Shouldn't `engine/lib.rs` export this type? In which case we also don't need `#[allow(dead_code)]` on any of its public mehtods.

Reply:
Resolved with #324.